### ISSUE TITLE
PR #4743 still doesn't work

### ISSUE THREAD

**Issue by evanchaoli** [Kong/kong#4814]
The PR #4743 was trying to fix the plugin `ldap-auth` to support LDAPS. Unfortunately the fix still has problem with the schema entity check:

    in 'plugins':
      - in entry 1 of 'plugins':
        in 'config':
          in 'start_tls': value must be false
          in '@entity':
        - in entry 1 of '@entity': 'ldaps' and 'start_tls' cannot be enabled simultaneously
    stack traceback:
        [C]: in function 'error'
       /usr/local/share/lua/5.1/kong/init.lua:382: in function 'init'
        init_by_lua:3: in main chunk

I configured `ldaps=true` and `start_tls=false`, which should pass the check, but it failed.

I tried to fix the `entity_checks`, however it's too hard to me. I don't understand the syntax of `entity_check`, and I couldn't find anything via Google search either.

The rest of logic of PR #4743 works fine if I comment out the `entity_checks`.

**Comment by kikito** [comment]
Hi, I'm sorry but I was not able to reproduce this issue. Are you using an older version of Kong, perhaps?

Here's the test I tried:

```
$ http -b :8001/services name=foo url=http://mockbin.com/request
{
    "client_certificate": null,
    "connect_timeout": 60000,
    "created_at": 1564581079,
    "host": "mockbin.com",
    "id": "abb9be9e-bced-419e-aec6-c9637670d4c7",
    "name": "foo",
    "path": "/request",
    "port": 80,
    "protocol": "http",
    "read_timeout": 60000,
    "retries": 5,
    "tags": null,
    "updated_at": 1564581079,
    "write_timeout": 60000
}

$ http -b :8001/services/foo/plugins name=ldap-auth \
   config:='{"ldap_host":"foo.com", "ldap_port":12345, "base_dn":"x", "attribute":"y", "start_tls": true, "ldaps": false}'
{
    "config": {
        "anonymous": null,
        "attribute": "y",
        "base_dn": "x",
        "cache_ttl": 60,
        "header_type": "ldap",
        "hide_credentials": false,
        "keepalive": 60000,
        "ldap_host": "foo.com",
        "ldap_port": 12345,
        "ldaps": false,
        "start_tls": true,
        "timeout": 10000,
        "verify_ldap_host": false
    },
    "consumer": null,
    "created_at": 1564581446,
    "enabled": true,
    "id": "3309a9f2-1198-4ccd-9dae-07f5dc258377",
    "name": "ldap-auth",
    "protocols": [
        "grpc",
        "grpcs",
        "http",
        "https"
    ],
    "route": null,
    "run_on": "first",
    "service": {
        "id": "abb9be9e-bced-419e-aec6-c9637670d4c7"
    },
    "tags": null
}
```

Could you include more info into this issue, like the exact requests you used, and your kong version?

**Comment by evanchaoli** [comment]
We are using Kong 1.2.

**Comment by evanchaoli** [comment]
We also verified that this fix works with Kong 1.3.
